eclipse/src/saros/filesystem/checksum/FileContentNotifierBridge.java

```diff
@@ -29,10 +40,22 @@ public FileContentNotifierBridge() {
         .addResourceChangeListener(this, IResourceChangeEvent.POST_CHANGE);
   }
 
+  private void initialize() {
+    SarosPluginContext.initComponent(this);
+  }
+
   @Override
   public void resourceChanged(IResourceChangeEvent event) {
     if (event.getDelta() == null) return;
 
+    if (sarosSessionManager == null) {
+      initialize();
```

Comment:
Nice hack ... circular dependency problem ?

Reply:
Yeah. Its not pretty, but it is already used in other places that had similar issues. (I know, not a good excuse.)

The problem is that the caching logic is required by the negotiation factory, which in turn is required by the session manager, which is required here for the caching logic.

This could be resolved by moving the caching logic into the session context and splitting the negotiation factory into a session negotiation factory (plugin context) and a resource negotiation factory (plugin context).

But, again, I don't think this should be done as part of this migration.

Reply:
Resolved in #1072.